python/pyspark/__init__.py

```diff
@@ -111,13 +124,13 @@ def keyword_only(func):
     """
 
     @wraps(func)
-    def wrapper(self, *args, **kwargs):
+    def wrapper(self: Any, *args: Any, **kwargs: Any) -> Any:
         if len(args) > 0:
             raise TypeError("Method %s forces keyword arguments." % func.__name__)
         self._input_kwargs = kwargs
         return func(self, **kwargs)
 
-    return wrapper
+    return cast(F, wrapper)
```

Comment:
Seems like the way to go, but might require some comment.

Reply:
Be we expect output is `F` type so I think we should just cast